src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -134,9 +134,9 @@ public boolean isOnStreetAllTheWay() {
     return isStreetOnly();
   }
 
-  /** TRUE if alt least one leg is a transit leg. */
+  /** TRUE if at least one leg is a transit leg. */
   public boolean hasTransit() {
-    return getTransitTimeSeconds() > 0;
+    return !getTransitDuration().isZero();
```

Comment:
It might be better to check if any of the legs is a transit leg. Now we get an error, if there is a only a single transit leg with a zero-time hop

Reply:
Do we want to include Flex legs in the check, like here: https://github.com/opentripplanner/OpenTripPlanner/pull/4335/commits/c4970a707e589eff861532313239f286b92790ea

Reply:
Looks good